opm/material/fluidsystems/blackoilpvt/Co2GasPvt.cpp

```diff
@@ -60,8 +55,15 @@ initFromState(const EclipseState& eclState, const Schedule&)
         OPM_THROW(std::runtime_error, "CO2STORE/CO2SOL can only be used with default values for STCOND!");
     }
 
-    gasReferenceDensity_[regionIdx] = CO2::gasDensity(T_ref, P_ref, extrapolate);
-    brineReferenceDensity_[regionIdx] = Brine::liquidDensity(T_ref, P_ref, salinity_[regionIdx], extrapolate);
+    std::size_t numRegions = eclState.runspec().tabdims().getNumPVTTables();
+    setNumRegions(numRegions);
+    for (size_t regionIdx = 0; regionIdx < numRegions; ++regionIdx) {
+        // Currently we only support constant salinity converted to mass fraction
+        salinity_[regionIdx] = eclState.getCo2StoreConfig().salinity();
+        gasReferenceDensity_[regionIdx] = CO2::gasDensity(T_ref, P_ref, extrapolate);
+        brineReferenceDensity_[regionIdx] = Brine::liquidDensity(T_ref, P_ref, salinity_[regionIdx], extrapolate);
```

Comment:
`brineReferenceDensity_` must be calculated with Ezrokhi's method when DENAQA is given in the deck